velox/exec/RowsStreamingWindowBuild.cpp

```diff
@@ -102,6 +102,15 @@ void RowsStreamingWindowBuild::noMoreInput() {
 
 std::shared_ptr<WindowPartition> RowsStreamingWindowBuild::nextPartition() {
   VELOX_CHECK(hasNextPartition());
+
+  if (outputPartition_ > 0) {
```

Comment:
This condition is there because you don't want to erase the first partition rightaway.

As I think more about it, I would rewrite this code so that if windowPartitions_ .size() > 2, then you now that you have a partition that has been output already. So then you clean it.

Reply:
I'd think we should use list or map. vector isn't good here. Or more efficient a ring vector to reuse the entry, instead of erase it.